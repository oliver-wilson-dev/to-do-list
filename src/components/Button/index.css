@import "../../styles/variables.css";

.button {
    padding: 10px 40px;
    border: 3px solid #f0134d;
    box-sizing: border-box;
    text-transform:uppercase;
    font-weight:400;
    text-align:center;
    transition: all 0.15s;
    font-size: 18px;
    background-color: var(--primaryBackgroundColour);
}
