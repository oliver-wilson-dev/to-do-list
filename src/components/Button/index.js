import React from 'react';
import PropTypes from 'prop-types';
import styles from './index.css';

const Button = ({ incrementCount }) => (
  <button className={styles.button} onClick={incrementCount} type="button">Click me!</button>
);

Button.propTypes = {
  incrementCount: PropTypes.func.isRequired
};

export default Button;
