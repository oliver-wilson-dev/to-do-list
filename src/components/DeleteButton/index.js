import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

const DeleteButton = ({ removeToDo, id }) => {
  const onClick = () => removeToDo({ id });

  return (
    <Button onClick={onClick}>
      <span role="img" aria-label="remove this todo">‚ùå</span>
    </Button>
  );
};

DeleteButton.propTypes = {
  removeToDo: PropTypes.func.isRequired,
  id: PropTypes.symbol.isRequired
};

export default DeleteButton;
