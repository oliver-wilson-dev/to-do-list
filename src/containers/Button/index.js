import { connect } from 'react-redux';
import Button from '../../components/Button';
import incrementCount from '../../state/actions/incrementCount';

const mapDispatchToProps = dispatch => ({
  incrementCount: () => dispatch(incrementCount())
});

export default connect(undefined, mapDispatchToProps)(Button);
