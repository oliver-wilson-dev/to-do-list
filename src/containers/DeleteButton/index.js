import { connect } from 'react-redux';
import DeleteButton from '../../components/DeleteButton';
import removeToDo from '../../state/actions/removeToDo';

const mapDispatchToProps = dispatch => ({
  removeToDo: ({ id }) => dispatch(removeToDo({ id }))
});

export default connect(undefined, mapDispatchToProps)(DeleteButton);
