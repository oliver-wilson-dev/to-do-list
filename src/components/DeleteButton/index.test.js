import React from 'react';
import { shallow } from 'enzyme';
import DeleteButton from './index';
import Button from '../Button';

jest.mock('../Button', () => {
  const Button = () => null;

  return Button;
});

const mockId = Symbol('test-id');
const defaultProps = { removeToDo: jest.fn(), id: mockId };

const render = overrideProps => shallow(<DeleteButton {...defaultProps} {...overrideProps} />);

describe('<DeleteButton/> component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  describe("when the button's onClick prop is called", () => {
    it("should call the removeToDo function with the task's id", () => {
      const mockRemoveToDo = jest.fn();
      const mockId = Symbol('test-id');

      render({ removeToDo: mockRemoveToDo, id: mockId })
        .find(Button).simulate('click');

      expect(mockRemoveToDo).toHaveBeenCalledWith({ id: mockId });
    });
  });
});
