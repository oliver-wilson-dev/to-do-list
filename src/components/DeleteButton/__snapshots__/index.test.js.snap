// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteButton/> component should render correctly 1`] = `
<Button
  onClick={[Function]}
>
  <span
    aria-label="remove this todo"
    role="img"
  >
    ‚ùå
  </span>
</Button>
`;
